Wasted/Pages/Place/OrdersPage.xaml.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Rg.Plugins.Popup.Services;
+using Wasted.Dummy_API.Business_Objects;
+using Wasted.DummyAPI.BusinessObjects;
+using Wasted.Utils;
+using Xamarin.Forms;
+using Xamarin.Forms.PancakeView;
+using Xamarin.Forms.Xaml;
+
+namespace Wasted.Pages.Place
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class OrdersPage : ContentPage
+    {
+        private DataService service;
+        public UserPlace CurrentUser { get; set; }
+        private IEnumerable<FoodPlace> ownedPlaces;
+        public IEnumerable<FoodPlace> OwnedPlaces
+        {
+            get { return ownedPlaces; }
+            set
+            {
+                ownedPlaces = value;
+                OnPropertyChanged();
+            }
+        }
+        public OrdersPage()
+        {
+            service = DependencyService.Get<DataService>();
+            CurrentUser = (UserPlace)service.CurrentUser;
+            OwnedPlaces = CurrentUser.OwnedPlaceIDs.Select(id => service.AllFoodPlaces[id - 1]);
+            InitializeComponent();
+
+            OrdersCollectionView.ItemsSource = service.OrderedDeals.Where(order =>
```

Comment:
HashSet<string> dealTitleHashset = new HashSet<string>(
                OwnedPlaces.SelectMany(
                        place => place.Deals.Select(deal => deal.Title)
                    )
                );
            OrdersCollectionView.ItemsSource =
                service.OrderedDeals.Where(order => dealTitleHashset.Contains(order.Title));
                
                Is O(n^2) complexity instead of O(n^3) and looks better